app/controllers/api/v1/bulk_download_controller.rb

```diff
@@ -106,13 +138,29 @@ def summary
         render json: @study_file_info
       end
 
+      def drs_info
+        drs_ids = params[:drs_ids]
+        file_info = Parallel.map(drs_ids, in_threads: 100) do |drs_id|
```

Comment:
I think this code already does what you want.  It accepts one or multiple ids, it will spawn up to 100 threads (but only 1 if only one id is requested).  And this method is synchronous.  Parallel.map will wait until all threads finish until it returns.  So the client code is really simple.  Jon and I talked about potentially expanding this to be a jobs-based system (right now, for the example with ~400 ids, it takes ~40 seconds for this method to return, so for larger batches we'll need a different approach)

Reply:
I've logged this as [feature request](https://broadworkbench.atlassian.net/browse/DR-1947) in the Data Repo board.